remove-webhook/remove-webhook.sh

```diff
@@ -57,17 +117,20 @@ then
 	kubectl config set clusters.local.insecure-skip-tls-verify true 
 fi
 
+rancher_version=$(kubectl get setting server-version -o jsonpath='{.value}')
+if [[ -z "$rancher_version" ]] || ! validate_rancher_version "$rancher_version"; then
```

Comment:
I'm not sure that this is working the way that you expect. When I run the following:

```
validate(){
return 1
}

[[ "v2.7.2" =~ 2\.7\.2 ]] || ! validate
```

I get 0 (success), but since the validate condition failed I would expect a `1` (or non-0 value). Can you elaborate more on this part?

Reply:
This is because the first test succeeds (pattern matches), so the second (!validate) doesn't need to run due to the OR.

Reply:
My bad on that one, but I also see same behavior on both failing:

```
[[ "v2.7.2" =~ 2\.7\.3 ]] || ! validate
```

Returns 0

Reply:
You're asserting that the first expression, the test, fails. Thus Bash tries the second expression due to the OR. The second one correctly asserts that it fails due to `!`.
Even `! validate` on its own returns zero, which makes me think it just reverses the return code somehow, interpreting it as bool depending on int value.

Reply:
My bad, you are right, thanks for guiding me through it.